aws-datastore/src/main/java/com/amplifyframework/datastore/network/MutationProcessor.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.network;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.datastore.DataStoreChannelEventName;
+import com.amplifyframework.datastore.storage.StorageItemChange;
+import com.amplifyframework.hub.HubChannel;
+import com.amplifyframework.hub.HubEvent;
+import com.amplifyframework.logging.Logger;
+
+import java.util.Objects;
+
+import io.reactivex.Single;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+
+/**
+ * The {@link MutationProcessor} observes the {@link MutationOutbox}, and publishes its items to an
+ * {@link AppSyncEndpoint}.
+ *
+ * The responses to these mutations are themselves forwarded to the Merger (TODO: write a merger.)
+ */
+final class MutationProcessor {
+    private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-datastore");
+
+    private final AppSyncEndpoint appSyncEndpoint;
```

Comment:
Minor nit - maybe this should be `GraphQLEndpoint` since we want this to be agnostic to AppSync

Reply:
In #271 I make an attempt to start sorting out the "this is appsync" from the "this is generic" -- However, I think I kept this as named (just) `AppSync`, since it has functionalities like Base/Delta sync and deliberately named mutations like Create Update Delete encoded into the interface.